vipps-checkout-api.md

```diff
@@ -164,39 +124,46 @@ Here is an example integration written in JavaScript that will make a request to
             return response.json();
           })
           .then(function (data) {
-            // Set token in URL and update the address field of the browser.
-            window.location.href = updateQueryStringParameter('token', data.token);
+            var vippsCheckout = VippsCheckout({
+              checkoutFrontendUrl: data.checkoutFrontendUrl,
+              iFrameContainerId: 'checkout-frame-container',
+              token: data.token
+            });
           })
           .catch(function (error) {
             console.error('Error:', error);
           });
       });
-
-      // Helper functions
-      function updateQueryStringParameter(key, value) {
-        var uri = window.location.href;
-        var re = new RegExp('([?&])' + key + '=.*?(&|$)', 'i');
-        var separator = uri.indexOf('?') !== -1 ? '&' : '?';
-        if (uri.match(re)) {
-          return uri.replace(re, '$1' + key + '=' + value + '$2');
-        } else {
-          return uri + separator + key + '=' + value;
-        }
-      }
-      function getParameterByName(name) {
-        var match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
-        var result = match && decodeURIComponent(match[1].replace(/\+/g, ' '));
-        return result;
-      }
     </script>
   </body>
 </html>
 
 ```
 
-Insert the token into the iframe and display it to the Customer. 
+`VippsCheckout` comes from the SDK that got loaded in `<head>` (async loading of SDK not available). The SDK's purpose is to attatch the iFrame to the given container element and load Vipps Checkout within it.
 
-The user then completes the session in the Iframe and gets redirected back to the fallBackUrl if applicable.
+The object argument to `VippsCheckout`
+```js
+{
+  checkoutFrontendUrl, // Will be supplied from our create session endpoint
+  iFrameContainerId, // The id of the html element to contain the Checkout iFrame
+  token // The token from create session endpoint that is specific to each checkout. (Optional when using token as queryParam flow as described below) 
+}
+```
+The SDK provides an alternative flow to enable the use of a query parameter to keep the session "sticky", e.g. if you refresh the page. If the query parameter `token` is present, and the token attribute in the argument object to VippsCheckout is not defined, the SDK will load the iFrame with the token from the query parameter.
+
+We provide a help method that when called will redirect to the current page but attatch a `token` queryParameter to the URL.
+Use it like this when receiving data from the create session endpoint to enable this feature.
+
+```js
+.then(function (data) {
```

Comment:
Can we be more explicit to which function invocation this promise handle refers to?